hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/validate/operators/common/HazelcastOperandTypeCheckerAware.java

```diff
@@ -73,6 +73,9 @@ default HazelcastCallBinding prepareBinding(SqlCallBinding binding) {
             operandTypeInference.inferOperandTypes(binding, binding.getValidator().getUnknownType(), operandTypes);
 
             for (int i = 0; i < binding.getOperandCount(); i++) {
+                if (binding.getOperandType(i).getSqlTypeName() != operandTypes[i].getSqlTypeName()) {
```

Comment:
This should not be there. Coercion is specific to operators and should not be enforced on the global level.